lib/host/wasi/vinode.cpp

```diff
@@ -105,15 +105,9 @@ WasiExpect<std::shared_ptr<VINode>> VINode::bind(__wasi_rights_t FRB,
 
 WasiExpect<void> VINode::pathCreateDirectory(std::shared_ptr<VINode> Fd,
                                              std::string_view Path) {
-  std::vector<char> Buffer;
-  if (auto Res = resolvePath(Fd, Path); unlikely(!Res)) {
```

Comment:
We should handle the special cases inside the `resolvePath` function. Removing the path checking is not the correct way to fix this issue.